static/src/javascripts-legacy/projects/commercial/modules/dfp/render-advert-label.js

```diff
@@ -1,14 +1,42 @@
 define([
     'Promise',
-    'common/utils/fastdom-promise'
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/views/svgs',
+    'text!commercial/views/ad-feedback-popup.html',
+    'common/utils/config'
 ], function (
     Promise,
-    fastdom
+    fastdom,
+    template,
+    svgs,
+    popupTemplate,
+    config
 ) {
     function renderAdvertLabel(adSlotNode) {
         if (shouldRenderLabel(adSlotNode)) {
+            var zIndexOverlay = 1050;
+            var feedbackPopup = '', feedbackThanksMessage = '';
+            if (config.switches.adFeedback) {
+                adSlotNode.style.zIndex = zIndexOverlay;
+                feedbackPopup = template(popupTemplate, {
```

Comment:
Any chance you can memoize that? I'm thinking:

- on first run, create a node `labelDiv` with all that stuff, just like you do below
- clone that node (with `deep` flag set to true) and set the ad slot specific data (i.e. just the id)
- insert that clone